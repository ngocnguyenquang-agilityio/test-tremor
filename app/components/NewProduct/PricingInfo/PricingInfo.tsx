"use client";

// Libs
import { Controller, useFormContext } from "react-hook-form";

// Components
import {
  TextInput,
  Text,
  Flex,
  Select,
  SelectItem,
  MultiSelect,
  MultiSelectItem,
} from "@tremor/react";

// Types
import { SelectOptionData } from "@/types";

// Constants
import { TYPE_PRICE, TAGS_PRICE, PRODUCT_TAGS } from "@/constants";

// Styles
import "@/styles/form.css";

const PricingInfo = () => {
  const { control } = useFormContext();

  const handleSend = () => {
    //TODO handle to check submit form with send button
  };

  return (
    <div className="w-full p-4 bg-white dark:bg-dark-tremor-primary rounded-lg w-[67%] shadow-box-icon-default">
      <Text className="text-primary dark:text-white font-bold text-xl mb-8">
        Pricing
      </Text>
      <Flex className="flex-col items-end">
        <Flex className="items-start flex-col sm:flex-row">
          <Flex className="flex-col">
            <Flex className="flex-col sm:flex-row">
              <Controller
                control={control}
                render={({ field }) => (
                  <div className="h-[70px] w-full md:max-w-[25%]">
                    <TextInput
                      id="price"
                      required
                      placeholder="Price"
                      className="w-full dark:text-white hover:bg-transparent bg-transparent dark:bg-transparent focus:bg-transparent rounded-b-none border-l-0 border-r-0 border-t-0 border-b-1 focus:border-b-2 focus:outline-none focus:border-tremor-brand-subtle dark:border-light dark:focus:border-white shadow-none hover:bg-transparent ring-0"
                      {...field}
                    />
                  </div>
                )}
                name="price"
              />
              <Controller
                control={control}
                render={() => (
                  <div className="h-[70px] mx-6 w-full md:max-w-[30%]">
                    <Select
                      placeholder="USD"
                      className="select-custom dark:text-white dark:border-light dark:focus:border-white">
                      {TYPE_PRICE.map((item: SelectOptionData) => (
                        <SelectItem key={item.value} value={item.value}>
                          {item.option}
                        </SelectItem>
                      ))}
                    </Select>
                  </div>
                )}
                name="type"
              />
              <Controller
                control={control}
                render={({ field }) => (
                  <div className="h-[70px] w-full">
                    <TextInput
                      id="sku"
                      placeholder="SKU"
                      className="w-full dark:text-white hover:bg-transparent bg-transparent dark:bg-transparent focus:bg-transparent rounded-b-none border-l-0 border-r-0 border-t-0 border-b-1 focus:border-b-2 focus:outline-none focus:border-tremor-brand-subtle dark:border-light dark:focus:border-white shadow-none hover:bg-transparent ring-0"
                      {...field}
                    />
                  </div>
                )}
                name="sku"
              />
            </Flex>

            <Controller
              control={control}
              render={() => (
                <div className="w-full mb-4">
                  <Text className="text-secondary dark:text-white">Tags</Text>
                  <MultiSelect
                    defaultValue={[
                      PRODUCT_TAGS.IN_STOCK,
                      PRODUCT_TAGS.OUT_OF_STOCK,
                    ]}
                    className="select-custom dark:text-white dark:border-light dark:focus:border-white">
                    {TAGS_PRICE.map((item: SelectOptionData) => (
                      <MultiSelectItem key={item.value} value={item.value}>
                        {item.option}
                      </MultiSelectItem>
                    ))}
                  </MultiSelect>
                </div>
              )}
              name="tags"
            />
          </Flex>
        </Flex>
      </Flex>
    </div>
  );
};

export default PricingInfo;
