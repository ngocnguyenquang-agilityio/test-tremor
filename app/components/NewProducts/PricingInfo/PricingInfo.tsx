"use client";

// Libs
import { useForm, Controller } from "react-hook-form";

// Components
import {
  TextInput,
  Button,
  Text,
  Flex,
  Select,
  SelectItem,
  MultiSelect,
  MultiSelectItem,
} from "@tremor/react";

// Types
import { TPricingInfo, SelectOptionData } from "@/types";

// Constants
import { TYPE_PRICE, TAGS_PRICE } from "@/constants";

// Styles
import "@/styles/form.css";

const PricingInfo = () => {
  const { control, handleSubmit } = useForm<TPricingInfo>({
    defaultValues: {
      price: "",
      type: "",
      sku: "",
      tags: [""],
    },
    mode: "onSubmit",
  });

  const handleSend = () => {
    //TODO handle to check submit form with send button
  };

  return (
    <form
      onSubmit={handleSubmit(handleSend)}
      className="w-full p-4 bg-white dark:bg-dark-tremor-primary rounded-lg w-[67%] shadow-box-icon-default">
      <Text className="text-primary dark:text-white font-bold text-xl mb-8">
        Pricing
      </Text>
      <Flex className="flex-col items-end">
        <Flex className="items-start flex-col sm:flex-row">
          <Flex className="flex-col">
            <Flex className="flex-col sm:flex-row">
              <Controller
                control={control}
                render={({ field }) => (
                  <div className="h-[70px] w-full md:max-w-[25%]">
                    <TextInput
                      id="price"
                      required
                      placeholder="Price"
                      className="w-full dark:text-white hover:bg-transparent bg-transparent dark:bg-transparent focus:bg-transparent rounded-b-none border-l-0 border-r-0 border-t-0 border-b-1 focus:border-b-2 focus:outline-none focus:border-tremor-brand-subtle dark:border-light dark:focus:border-white shadow-none hover:bg-transparent ring-0"
                      {...field}
                    />
                  </div>
                )}
                name="price"
              />
              <Controller
                control={control}
                render={() => (
                  <div className="h-[70px] mx-6 w-full md:max-w-[30%]">
                    <Select
                      placeholder="USD"
                      className="select-custom dark:text-white dark:border-light dark:focus:border-white">
                      {TYPE_PRICE.map((item: SelectOptionData) => (
                        <SelectItem key={item.value} value={item.value}>
                          {item.option}
                        </SelectItem>
                      ))}
                    </Select>
                  </div>
                )}
                name="type"
              />
              <Controller
                control={control}
                render={({ field }) => (
                  <div className="h-[70px] w-full">
                    <TextInput
                      id="sku"
                      placeholder="SKU"
                      className="w-full dark:text-white hover:bg-transparent bg-transparent dark:bg-transparent focus:bg-transparent rounded-b-none border-l-0 border-r-0 border-t-0 border-b-1 focus:border-b-2 focus:outline-none focus:border-tremor-brand-subtle dark:border-light dark:focus:border-white shadow-none hover:bg-transparent ring-0"
                      {...field}
                    />
                  </div>
                )}
                name="sku"
              />
            </Flex>
            <Flex className="flex-col items-start">
              <Text className="text-secondary mb-3 dark:text-white">Tags</Text>
              <Controller
                control={control}
                render={() => (
                  <div className="h-[70px] w-full">
                    <MultiSelect
                      defaultValue={["In Stock", "Out of Stock"]}
                      className="select-custom dark:text-white dark:border-light dark:focus:border-white">
                      {TAGS_PRICE.map((item: SelectOptionData) => (
                        <MultiSelectItem key={item.value} value={item.value}>
                          {item.option}
                        </MultiSelectItem>
                      ))}
                    </MultiSelect>
                  </div>
                )}
                name="tags"
              />
            </Flex>
          </Flex>
        </Flex>
        <Button
          type="submit"
          className="antialiased min-w-[80px] py-[12px] text-center uppercase sm:px-[21px] bg-gradient-primary dark:bg-gradient-pickled !shadow-btn-primary px-6 py-2.5 rounded-lg border-0 hover:!shadow-btn-primary-hover"
          size="xs">
          <Text className="items-center uppercase py-[2px] text-xs font-bold font-primary text-white dark:text-dark-tremor-content-title">
            Send
          </Text>
        </Button>
      </Flex>
    </form>
  );
};

export default PricingInfo;
