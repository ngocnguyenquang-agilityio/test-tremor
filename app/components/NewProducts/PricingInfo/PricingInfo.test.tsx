import { render } from "@testing-library/react";

// Components
import PricingInfo from "./PricingInfo";

describe("Testing PricingInfo component", () => {
  it("Should match snapshot", () => {
    const component = render(<PricingInfo />);
    expect(component).toMatchSnapshot();
  });
});
