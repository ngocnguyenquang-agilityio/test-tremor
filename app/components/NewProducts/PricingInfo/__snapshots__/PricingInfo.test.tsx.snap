// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing PricingInfo component Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="w-full p-4 bg-white dark:bg-dark-tremor-primary rounded-lg w-[67%] shadow-box-icon-default"
      >
        <p
          class="text-tremor-default text-primary dark:text-white font-bold text-xl mb-8"
        >
          Pricing
        </p>
        <div
          class="tremor-Flex-root flex w-full justify-between flex-col items-end"
        >
          <div
            class="tremor-Flex-root flex w-full justify-between items-start flex-col sm:flex-row"
          >
            <div
              class="tremor-Flex-root flex w-full justify-between items-center flex-col"
            >
              <div
                class="tremor-Flex-root flex w-full justify-between items-center flex-col sm:flex-row"
              >
                <div
                  class="h-[70px] w-full md:max-w-[25%]"
                >
                  <div
                    class="tremor-TextInput-root relative flex items-center min-w-[10rem] outline-none rounded-tremor-default transition duration-100 shadow-tremor-input dark:shadow-dark-tremor-input dark:hover:bg-dark-tremor-background-muted text-tremor-content border-tremor-border border w-full dark:text-white bg-transparent dark:bg-transparent focus:bg-transparent rounded-b-none border-l-0 border-r-0 border-t-0 border-b-1 focus:border-b-2 focus:outline-none focus:border-tremor-brand-subtle dark:border-light dark:focus:border-white shadow-none hover:bg-transparent ring-0"
                  >
                    <input
                      class="tremor-TextInput-input w-full focus:outline-none focus:ring-0 border-none bg-transparent text-tremor-default rounded-tremor-default transition duration-100 text-tremor-content-emphasis dark:text-dark-tremor-content-emphasis [appearance:textfield] [&::-webkit-outer-spin-button]:appearance-none [&::-webkit-inner-spin-button]:appearance-none pl-3 pr-4 py-2 placeholder:text-tremor-content dark:placeholder:text-dark-tremor-content"
                      data-testid="base-input"
                      id="price"
                      name="price"
                      placeholder="Price"
                      required=""
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="h-[70px] mx-6 w-full md:max-w-[30%]"
                >
                  <div
                    class="w-full min-w-[10rem] relative text-tremor-default select-custom dark:text-white dark:border-light dark:focus:border-white"
                    data-headlessui-state=""
                  >
                    <button
                      aria-expanded="false"
                      aria-haspopup="listbox"
                      class="w-full outline-none text-left whitespace-nowrap truncate rounded-tremor-default focus:ring-2 transition duration-100 shadow-tremor-input focus:border-tremor-brand-subtle focus:ring-tremor-brand-muted dark:shadow-dark-tremor-input dark:focus:border-dark-tremor-brand-subtle dark:focus:ring-dark-tremor-brand-muted pl-3 pr-8 py-2 border bg-tremor-background dark:bg-dark-tremor-background hover:bg-tremor-background-muted dark:hover:bg-dark-tremor-background-muted text-tremor-content dark:text-dark-tremor-content border-tremor-border dark:border-dark-tremor-border"
                      data-headlessui-state=""
                      id="headlessui-listbox-button-:r0:"
                      type="button"
                    >
                      <span
                        class="w-[90%] block truncate"
                      >
                        USD
                      </span>
                      <span
                        class="absolute inset-y-0 right-0 flex items-center mr-3"
                      >
                        <svg
                          class="tremor-Select-arrowDownIcon flex-none text-tremor-content-subtle dark:text-dark-tremor-content-subtle h-4 w-4"
                          fill="none"
                          stroke="currentColor"
                          stroke-width="2.5"
                          viewBox="0 0 24 24"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M19 9l-7 7-7-7"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                          />
                        </svg>
                      </span>
                    </button>
                  </div>
                </div>
                <div
                  class="h-[70px] w-full"
                >
                  <div
                    class="tremor-TextInput-root relative flex items-center min-w-[10rem] outline-none rounded-tremor-default transition duration-100 shadow-tremor-input dark:shadow-dark-tremor-input dark:hover:bg-dark-tremor-background-muted text-tremor-content border-tremor-border border w-full dark:text-white bg-transparent dark:bg-transparent focus:bg-transparent rounded-b-none border-l-0 border-r-0 border-t-0 border-b-1 focus:border-b-2 focus:outline-none focus:border-tremor-brand-subtle dark:border-light dark:focus:border-white shadow-none hover:bg-transparent ring-0"
                  >
                    <input
                      class="tremor-TextInput-input w-full focus:outline-none focus:ring-0 border-none bg-transparent text-tremor-default rounded-tremor-default transition duration-100 text-tremor-content-emphasis dark:text-dark-tremor-content-emphasis [appearance:textfield] [&::-webkit-outer-spin-button]:appearance-none [&::-webkit-inner-spin-button]:appearance-none pl-3 pr-4 py-2 placeholder:text-tremor-content dark:placeholder:text-dark-tremor-content"
                      data-testid="base-input"
                      id="sku"
                      name="sku"
                      placeholder="SKU"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
              <div
                class="tremor-Flex-root flex w-full justify-between flex-col items-start"
              >
                <p
                  class="text-tremor-default text-secondary mb-3 dark:text-white"
                >
                  Tags
                </p>
                <div
                  class="h-[70px] w-full"
                >
                  <div
                    class="w-full min-w-[10rem] relative text-tremor-default select-custom dark:text-white dark:border-light dark:focus:border-white"
                    data-headlessui-state=""
                  >
                    <button
                      aria-expanded="false"
                      aria-haspopup="listbox"
                      class="w-full outline-none text-left whitespace-nowrap truncate rounded-tremor-default focus:ring-2 transition duration-100 shadow-tremor-input focus:border-tremor-brand-subtle focus:ring-tremor-brand-muted dark:shadow-dark-tremor-input dark:focus:border-dark-tremor-brand-subtle dark:focus:ring-dark-tremor-brand-muted pl-3 pr-8 py-1.5 border bg-tremor-background dark:bg-dark-tremor-background hover:bg-tremor-background-muted dark:hover:bg-dark-tremor-background-muted text-tremor-content-emphasis dark:text-dark-tremor-content-emphasis border-tremor-border dark:border-dark-tremor-border"
                      data-headlessui-state=""
                      id="headlessui-listbox-button-:r1:"
                      type="button"
                    >
                      <div
                        class="h-6 flex items-center"
                      >
                        <div
                          class="flex flex-nowrap overflow-x-scroll [&::-webkit-scrollbar]:hidden [scrollbar-width:none] gap-x-1 mr-5 -ml-1.5 relative"
                        >
                          <div
                            class="max-w-[100px] lg:max-w-[200px] flex justify-center items-center pl-2 pr-1.5 py-1 font-medium rounded-tremor-small bg-tremor-background-subtle dark:bg-dark-tremor-background-subtle text-tremor-content-emphasis dark:text-dark-tremor-content-emphasis"
                          >
                            <div
                              class="text-xs truncate "
                            >
                              Out of Stock
                            </div>
                            <div>
                              <svg
                                class="tremor-MultiSelect-clearIconItem cursor-pointer rounded-tremor-full w-3.5 h-3.5 ml-2 text-tremor-content-subtle hover:text-tremor-content dark:text-dark-tremor-content-subtle dark:hover:text-tremor-content"
                                fill="none"
                                height="100%"
                                stroke="currentColor"
                                stroke-linecap="round"
                                stroke-linejoin="round"
                                stroke-width="2"
                                viewBox="0 0 24 24"
                                width="100%"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <line
                                  x1="18"
                                  x2="6"
                                  y1="6"
                                  y2="18"
                                />
                                <line
                                  x1="6"
                                  x2="18"
                                  y1="6"
                                  y2="18"
                                />
                              </svg>
                            </div>
                          </div>
                          <div
                            class="max-w-[100px] lg:max-w-[200px] flex justify-center items-center pl-2 pr-1.5 py-1 font-medium rounded-tremor-small bg-tremor-background-subtle dark:bg-dark-tremor-background-subtle text-tremor-content-emphasis dark:text-dark-tremor-content-emphasis"
                          >
                            <div
                              class="text-xs truncate "
                            >
                              In Stock
                            </div>
                            <div>
                              <svg
                                class="tremor-MultiSelect-clearIconItem cursor-pointer rounded-tremor-full w-3.5 h-3.5 ml-2 text-tremor-content-subtle hover:text-tremor-content dark:text-dark-tremor-content-subtle dark:hover:text-tremor-content"
                                fill="none"
                                height="100%"
                                stroke="currentColor"
                                stroke-linecap="round"
                                stroke-linejoin="round"
                                stroke-width="2"
                                viewBox="0 0 24 24"
                                width="100%"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <line
                                  x1="18"
                                  x2="6"
                                  y1="6"
                                  y2="18"
                                />
                                <line
                                  x1="6"
                                  x2="18"
                                  y1="6"
                                  y2="18"
                                />
                              </svg>
                            </div>
                          </div>
                        </div>
                      </div>
                      <span
                        class="absolute inset-y-0 right-0 flex items-center mr-2.5"
                      >
                        <svg
                          class="tremor-MultiSelect-arrowDownIcon flex-none text-tremor-content-subtle dark:text-dark-tremor-content-subtle h-4 w-4"
                          fill="none"
                          stroke="currentColor"
                          stroke-width="2.5"
                          viewBox="0 0 24 24"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M19 9l-7 7-7-7"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                          />
                        </svg>
                      </span>
                    </button>
                    <button
                      class="absolute inset-y-0 right-0 flex items-center mr-8"
                      type="button"
                    >
                      <svg
                        class="tremor-MultiSelect-clearIconAllItems flex-none text-tremor-content-subtle dark:text-dark-tremor-content-subtle h-4 w-4"
                        fill="currentColor"
                        viewBox="0 0 20 20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          clip-rule="evenodd"
                          d="M10 18a8 8 0 100-16 8 8 0 000 16zM8.707 7.293a1 1 0 00-1.414 1.414L8.586 10l-1.293 1.293a1 1 0 101.414 1.414L10 11.414l1.293 1.293a1 1 0 001.414-1.414L11.414 10l1.293-1.293a1 1 0 00-1.414-1.414L10 8.586 8.707 7.293z"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <button
            class="tremor-Button-root flex-shrink-0 inline-flex justify-center items-center group font-medium outline-none rounded-tremor-default shadow-tremor-input dark:shadow-dark-tremor-input text-xs border-tremor-brand dark:border-dark-tremor-brand text-tremor-brand-inverted dark:text-dark-tremor-brand-inverted hover:bg-tremor-brand-emphasis dark:hover:bg-dark-tremor-brand-emphasis hover:border-tremor-brand-emphasis dark:hover:border-dark-tremor-brand-emphasis antialiased min-w-[80px] text-center uppercase sm:px-[21px] bg-gradient-primary dark:bg-gradient-pickled !shadow-btn-primary px-6 py-2.5 rounded-lg border-0 hover:!shadow-btn-primary-hover"
            type="submit"
          >
            <span
              class="tremor-Button-text text-sm whitespace-nowrap"
            >
              <p
                class="text-tremor-default items-center uppercase py-[2px] text-xs font-bold font-primary text-white dark:text-dark-tremor-content-title"
              >
                Send
              </p>
            </span>
          </button>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="w-full p-4 bg-white dark:bg-dark-tremor-primary rounded-lg w-[67%] shadow-box-icon-default"
    >
      <p
        class="text-tremor-default text-primary dark:text-white font-bold text-xl mb-8"
      >
        Pricing
      </p>
      <div
        class="tremor-Flex-root flex w-full justify-between flex-col items-end"
      >
        <div
          class="tremor-Flex-root flex w-full justify-between items-start flex-col sm:flex-row"
        >
          <div
            class="tremor-Flex-root flex w-full justify-between items-center flex-col"
          >
            <div
              class="tremor-Flex-root flex w-full justify-between items-center flex-col sm:flex-row"
            >
              <div
                class="h-[70px] w-full md:max-w-[25%]"
              >
                <div
                  class="tremor-TextInput-root relative flex items-center min-w-[10rem] outline-none rounded-tremor-default transition duration-100 shadow-tremor-input dark:shadow-dark-tremor-input dark:hover:bg-dark-tremor-background-muted text-tremor-content border-tremor-border border w-full dark:text-white bg-transparent dark:bg-transparent focus:bg-transparent rounded-b-none border-l-0 border-r-0 border-t-0 border-b-1 focus:border-b-2 focus:outline-none focus:border-tremor-brand-subtle dark:border-light dark:focus:border-white shadow-none hover:bg-transparent ring-0"
                >
                  <input
                    class="tremor-TextInput-input w-full focus:outline-none focus:ring-0 border-none bg-transparent text-tremor-default rounded-tremor-default transition duration-100 text-tremor-content-emphasis dark:text-dark-tremor-content-emphasis [appearance:textfield] [&::-webkit-outer-spin-button]:appearance-none [&::-webkit-inner-spin-button]:appearance-none pl-3 pr-4 py-2 placeholder:text-tremor-content dark:placeholder:text-dark-tremor-content"
                    data-testid="base-input"
                    id="price"
                    name="price"
                    placeholder="Price"
                    required=""
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class="h-[70px] mx-6 w-full md:max-w-[30%]"
              >
                <div
                  class="w-full min-w-[10rem] relative text-tremor-default select-custom dark:text-white dark:border-light dark:focus:border-white"
                  data-headlessui-state=""
                >
                  <button
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    class="w-full outline-none text-left whitespace-nowrap truncate rounded-tremor-default focus:ring-2 transition duration-100 shadow-tremor-input focus:border-tremor-brand-subtle focus:ring-tremor-brand-muted dark:shadow-dark-tremor-input dark:focus:border-dark-tremor-brand-subtle dark:focus:ring-dark-tremor-brand-muted pl-3 pr-8 py-2 border bg-tremor-background dark:bg-dark-tremor-background hover:bg-tremor-background-muted dark:hover:bg-dark-tremor-background-muted text-tremor-content dark:text-dark-tremor-content border-tremor-border dark:border-dark-tremor-border"
                    data-headlessui-state=""
                    id="headlessui-listbox-button-:r0:"
                    type="button"
                  >
                    <span
                      class="w-[90%] block truncate"
                    >
                      USD
                    </span>
                    <span
                      class="absolute inset-y-0 right-0 flex items-center mr-3"
                    >
                      <svg
                        class="tremor-Select-arrowDownIcon flex-none text-tremor-content-subtle dark:text-dark-tremor-content-subtle h-4 w-4"
                        fill="none"
                        stroke="currentColor"
                        stroke-width="2.5"
                        viewBox="0 0 24 24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M19 9l-7 7-7-7"
                          stroke-linecap="round"
                          stroke-linejoin="round"
                        />
                      </svg>
                    </span>
                  </button>
                </div>
              </div>
              <div
                class="h-[70px] w-full"
              >
                <div
                  class="tremor-TextInput-root relative flex items-center min-w-[10rem] outline-none rounded-tremor-default transition duration-100 shadow-tremor-input dark:shadow-dark-tremor-input dark:hover:bg-dark-tremor-background-muted text-tremor-content border-tremor-border border w-full dark:text-white bg-transparent dark:bg-transparent focus:bg-transparent rounded-b-none border-l-0 border-r-0 border-t-0 border-b-1 focus:border-b-2 focus:outline-none focus:border-tremor-brand-subtle dark:border-light dark:focus:border-white shadow-none hover:bg-transparent ring-0"
                >
                  <input
                    class="tremor-TextInput-input w-full focus:outline-none focus:ring-0 border-none bg-transparent text-tremor-default rounded-tremor-default transition duration-100 text-tremor-content-emphasis dark:text-dark-tremor-content-emphasis [appearance:textfield] [&::-webkit-outer-spin-button]:appearance-none [&::-webkit-inner-spin-button]:appearance-none pl-3 pr-4 py-2 placeholder:text-tremor-content dark:placeholder:text-dark-tremor-content"
                    data-testid="base-input"
                    id="sku"
                    name="sku"
                    placeholder="SKU"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
            <div
              class="tremor-Flex-root flex w-full justify-between flex-col items-start"
            >
              <p
                class="text-tremor-default text-secondary mb-3 dark:text-white"
              >
                Tags
              </p>
              <div
                class="h-[70px] w-full"
              >
                <div
                  class="w-full min-w-[10rem] relative text-tremor-default select-custom dark:text-white dark:border-light dark:focus:border-white"
                  data-headlessui-state=""
                >
                  <button
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    class="w-full outline-none text-left whitespace-nowrap truncate rounded-tremor-default focus:ring-2 transition duration-100 shadow-tremor-input focus:border-tremor-brand-subtle focus:ring-tremor-brand-muted dark:shadow-dark-tremor-input dark:focus:border-dark-tremor-brand-subtle dark:focus:ring-dark-tremor-brand-muted pl-3 pr-8 py-1.5 border bg-tremor-background dark:bg-dark-tremor-background hover:bg-tremor-background-muted dark:hover:bg-dark-tremor-background-muted text-tremor-content-emphasis dark:text-dark-tremor-content-emphasis border-tremor-border dark:border-dark-tremor-border"
                    data-headlessui-state=""
                    id="headlessui-listbox-button-:r1:"
                    type="button"
                  >
                    <div
                      class="h-6 flex items-center"
                    >
                      <div
                        class="flex flex-nowrap overflow-x-scroll [&::-webkit-scrollbar]:hidden [scrollbar-width:none] gap-x-1 mr-5 -ml-1.5 relative"
                      >
                        <div
                          class="max-w-[100px] lg:max-w-[200px] flex justify-center items-center pl-2 pr-1.5 py-1 font-medium rounded-tremor-small bg-tremor-background-subtle dark:bg-dark-tremor-background-subtle text-tremor-content-emphasis dark:text-dark-tremor-content-emphasis"
                        >
                          <div
                            class="text-xs truncate "
                          >
                            Out of Stock
                          </div>
                          <div>
                            <svg
                              class="tremor-MultiSelect-clearIconItem cursor-pointer rounded-tremor-full w-3.5 h-3.5 ml-2 text-tremor-content-subtle hover:text-tremor-content dark:text-dark-tremor-content-subtle dark:hover:text-tremor-content"
                              fill="none"
                              height="100%"
                              stroke="currentColor"
                              stroke-linecap="round"
                              stroke-linejoin="round"
                              stroke-width="2"
                              viewBox="0 0 24 24"
                              width="100%"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <line
                                x1="18"
                                x2="6"
                                y1="6"
                                y2="18"
                              />
                              <line
                                x1="6"
                                x2="18"
                                y1="6"
                                y2="18"
                              />
                            </svg>
                          </div>
                        </div>
                        <div
                          class="max-w-[100px] lg:max-w-[200px] flex justify-center items-center pl-2 pr-1.5 py-1 font-medium rounded-tremor-small bg-tremor-background-subtle dark:bg-dark-tremor-background-subtle text-tremor-content-emphasis dark:text-dark-tremor-content-emphasis"
                        >
                          <div
                            class="text-xs truncate "
                          >
                            In Stock
                          </div>
                          <div>
                            <svg
                              class="tremor-MultiSelect-clearIconItem cursor-pointer rounded-tremor-full w-3.5 h-3.5 ml-2 text-tremor-content-subtle hover:text-tremor-content dark:text-dark-tremor-content-subtle dark:hover:text-tremor-content"
                              fill="none"
                              height="100%"
                              stroke="currentColor"
                              stroke-linecap="round"
                              stroke-linejoin="round"
                              stroke-width="2"
                              viewBox="0 0 24 24"
                              width="100%"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <line
                                x1="18"
                                x2="6"
                                y1="6"
                                y2="18"
                              />
                              <line
                                x1="6"
                                x2="18"
                                y1="6"
                                y2="18"
                              />
                            </svg>
                          </div>
                        </div>
                      </div>
                    </div>
                    <span
                      class="absolute inset-y-0 right-0 flex items-center mr-2.5"
                    >
                      <svg
                        class="tremor-MultiSelect-arrowDownIcon flex-none text-tremor-content-subtle dark:text-dark-tremor-content-subtle h-4 w-4"
                        fill="none"
                        stroke="currentColor"
                        stroke-width="2.5"
                        viewBox="0 0 24 24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M19 9l-7 7-7-7"
                          stroke-linecap="round"
                          stroke-linejoin="round"
                        />
                      </svg>
                    </span>
                  </button>
                  <button
                    class="absolute inset-y-0 right-0 flex items-center mr-8"
                    type="button"
                  >
                    <svg
                      class="tremor-MultiSelect-clearIconAllItems flex-none text-tremor-content-subtle dark:text-dark-tremor-content-subtle h-4 w-4"
                      fill="currentColor"
                      viewBox="0 0 20 20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M10 18a8 8 0 100-16 8 8 0 000 16zM8.707 7.293a1 1 0 00-1.414 1.414L8.586 10l-1.293 1.293a1 1 0 101.414 1.414L10 11.414l1.293 1.293a1 1 0 001.414-1.414L11.414 10l1.293-1.293a1 1 0 00-1.414-1.414L10 8.586 8.707 7.293z"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
        <button
          class="tremor-Button-root flex-shrink-0 inline-flex justify-center items-center group font-medium outline-none rounded-tremor-default shadow-tremor-input dark:shadow-dark-tremor-input text-xs border-tremor-brand dark:border-dark-tremor-brand text-tremor-brand-inverted dark:text-dark-tremor-brand-inverted hover:bg-tremor-brand-emphasis dark:hover:bg-dark-tremor-brand-emphasis hover:border-tremor-brand-emphasis dark:hover:border-dark-tremor-brand-emphasis antialiased min-w-[80px] text-center uppercase sm:px-[21px] bg-gradient-primary dark:bg-gradient-pickled !shadow-btn-primary px-6 py-2.5 rounded-lg border-0 hover:!shadow-btn-primary-hover"
          type="submit"
        >
          <span
            class="tremor-Button-text text-sm whitespace-nowrap"
          >
            <p
              class="text-tremor-default items-center uppercase py-[2px] text-xs font-bold font-primary text-white dark:text-dark-tremor-content-title"
            >
              Send
            </p>
          </span>
        </button>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
